src/general/Mutations.cs

```diff
@@ -441,6 +418,35 @@ private Color RandomColour(float opaqueness)
             opaqueness);
     }
 
+    /// <summary>
+    ///   Used to determine if a newly mutated species needs to be in a different genus.
+    /// </summary>
+    /// <param name="species1">The first species. Function is not order-dependent.</param>
+    /// <param name="species2">The second species. Function is not order-dependent.</param>
+    /// <returns>True if the two species should be a new genus, false otherwise.</returns>
+    private bool NewGenus(MicrobeSpecies species1, MicrobeSpecies species2)
+    {
+        var species1UniqueOrganelles = new HashSet<string>();
+        foreach (var organelle in species1.Organelles.Organelles)
+        {
+            if (!species1UniqueOrganelles.Contains(organelle.Definition.Name))
+            {
+                species1UniqueOrganelles.Add(organelle.Definition.Name);
+            }
+        }
```

Comment:
I think this whole thing could be a single LINQ statement (also using Definition instances instead of name strings, as I think that'll be better to just use reference comparisons than string comparisons).

`species1.Organelles.Select(o => o.Definition).ToHashSet()`
https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.tohashset?view=net-5.0

I didn't try so might need to same AsEnumerable, imports etc. tweaking.

Reply:
Very useful; I was wondering how to make functional calls in C#.